public/app/core/components/RolePicker/UserRolePicker.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { FC } from 'react';
+import { getBackendSrv } from '@grafana/runtime';
+import { Role, OrgRole } from 'app/types';
+import { RolePicker } from './RolePicker';
+
+export interface Props {
+  builtInRole: OrgRole;
+  userId: number;
+  orgId?: number;
+  onBuiltinRoleChange: (newRole: OrgRole) => void;
+  disabled?: boolean;
+}
+
+export const UserRolePicker: FC<Props> = ({ builtInRole, userId, orgId, onBuiltinRoleChange, disabled }) => {
+  return (
+    <RolePicker
+      builtInRole={builtInRole}
+      onRolesChange={(roles) => updateUserRoles(roles, userId, orgId)}
+      onBuiltinRoleChange={onBuiltinRoleChange}
+      getRoleOptions={async () => getRolesOptions(orgId)}
+      getRoles={async () => getUserRoles(userId, orgId)}
+      getBuiltinRoles={async () => getBuiltinRoles(orgId)}
```

Comment:
I was thinking if it'd be better that `RolePicker` accepts the resolved values for these promises instead. It'd simplify the component, as it wouldn't need to make those api calls and could use the values directly from the props. Not sure if it's better than the current approach though.